utils/utils.tsx

```diff
@@ -234,58 +236,59 @@ export function hasSoundOptions() {
     return (!UserAgent.isEdge());
 }
 
-export function getDateForUnixTicks(ticks) {
+export function getDateForUnixTicks(ticks: number): Date {
     return new Date(ticks);
 }
 
 // returns Unix timestamp in milliseconds
-export function getTimestamp() {
+export function getTimestamp(): number {
     return Date.now();
 }
 
-export function getRemainingDaysFromFutureTimestamp(timestamp) {
+export function getRemainingDaysFromFutureTimestamp(timestamp?: number): number {
     const MS_PER_DAY = 24 * 60 * 60 * 1000;
-    const futureDate = new Date(timestamp);
+    const futureDate = new Date(timestamp as number);
     const utcFuture = Date.UTC(futureDate.getFullYear(), futureDate.getMonth(), futureDate.getDate());
     const today = new Date();
     const utcToday = Date.UTC(today.getFullYear(), today.getMonth(), today.getDate());
 
     return Math.floor((utcFuture - utcToday) / MS_PER_DAY);
 }
 
-export function getLocaleDateFromUTC(timestamp, format = 'YYYY/MM/DD HH:mm:ss', userTimezone = '') {
+export function getLocaleDateFromUTC(timestamp: number, format = 'YYYY/MM/DD HH:mm:ss', userTimezone = '') {
     if (!timestamp) {
         return moment.now();
     }
     const timezone = userTimezone ? ' ' + moment().tz(userTimezone).format('z') : '';
     return moment.unix(timestamp).format(format) + timezone;
 }
 
-export function replaceHtmlEntities(text) {
-    var tagsToReplace = {
+export function replaceHtmlEntities(text: string) {
+    const tagsToReplace = {
         '&amp;': '&',
         '&lt;': '<',
         '&gt;': '>',
     };
-    var newtext = text;
-    for (var tag in tagsToReplace) {
+    let newtext = text;
+    for (const tag in tagsToReplace) {
+        // @ts-expect-error // TODO: resolve this typing and see if using function this is necessary
```

Comment:
For this and the other 3 `@ts-expect-error` annotations in this file, I think the longer term fix is remove the usage of `this`, if possible.

Reply:
Maybe we could fix this by switching the loop to a for-of instead of a for-in? That should get rid of the `hasOwnPropery` call at least since that's always miserable to use